### PR TITLE
Allow gone/redirect replacement.

### PR DESCRIPTION
**Pull request by elliotcm**
This allows gones and redirects to replace content
and also allows content to replace gones and redirects
for republishing.

It does not allow normal content to overwrite other
normal content, reducing the chance of accidentally
blatting something important.

Half of https://trello.com/c/MtnVvpID/420-allow-overwriting-of-redirects-and-gones

### CODE REVIEW

File: app/commands/put_content_with_links.rb [hunk 1]
```diff
@@ -48,6 +48,8 @@ def create_or_update_live_content_item!(draft_content_item)
       )
 
       LiveContentItem.create_or_replace(attributes) do |item|
+        clear_space_for(item)
```
Comment (danielroseman) [inline app/commands/put_content_with_links.rb]: Seems to me that we only need to do this when item itself is new, so maybe we can do `clear_space_for(item) if item.new_record?`

Comment (elliotcm) [inline app/commands/put_content_with_links.rb]: True.  Updated.

File: spec/commands/v2/put_content_spec.rb [hunk 2]
```diff
@@ -59,5 +58,83 @@
...
+
+    context "when a content item wants to replace a content item" do
+      before do
+        create(:draft_content_item, base_path: base_path)
+      end
+
+      it "does not replace the existing content" do
+        expect {
+          described_class.call(payload)
+        }.to raise_error(CommandRetryableError)
```
Comment (danielroseman) [inline spec/commands/v2/put_content_spec.rb]: Not part of this change necessarily, but we do probably want to return a different kind of error here; there's no point in retrying this call, it will never succeed.

**Comment by danielroseman** [comment]
We need to do this in v1/put_content_with_links as well I think.

File: spec/commands/v2/put_content_spec.rb (hunk 2 above)
Comment (elliotcm) [inline spec/commands/v2/put_content_spec.rb]: Yeah, we can update this when it's changed.  The test will pick up the change.

**Comment by elliotcm** [comment]
@danielroseman Already is.  I opted not to do specific tests for the v1 endpoints because the business logic is shared with the (fully tested) v2 endpoint, and the v1s are not long for this world.

**Comment by danielroseman** [comment]
@elliotcm yes sorry, realised that straight after I made the comment and deleted it, but that doesn't unsend the notification...
